### PR TITLE
Make sure integration tests exits immediately

### PR DESCRIPTION
**Pull request by boxhock**
## Changes

- Add timeout to CI integration tests
- Add shutdown event to cache warmer (cancels timers)
- Fix a failing test for synthetix-debt-pool

## Steps to Test

1. Run `yarn test:ci:integration` and make sure it exits cleanly as soon as tests are completed

## Quality Assurance

- [ ] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `<ADAPTER_PACKAGE>/schemas/env.json` and `<ADAPTER_PACKAGE>/README.md`
- [ ] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `infra-k8s` configuration file.
- [x] The branch naming follows git flow (`feature/x`, `chore/x`, `release/x`, `hotfix/x`, `fix/x`) or is created from Clubhouse/Shortcut
- [x] This is related to a maximum of one Clubhouse/Shortcut story or GitHub issue
- [x] Types are safe (avoid TypeScript/TSLint features like any and disable, instead use more specific types)

### CODE REVIEW

File: packages/core/bootstrap/src/lib/cache-warmer/epics.ts [hunk 1]
```diff
@@ -177,17 +181,20 @@ export const warmupSubscriber: Epic<AnyAction, AnyAction, any, EpicDependencies>
...
         // unsubscribe our warmup algo when a matching unsubscribe comes in
         takeUntil(
           action$.pipe(
             filter(warmupUnsubscribed.match || warmupStopped.match),
-            filter((a) => a.payload.key === key),
+            filter(
+              (a: PayloadAction<WarmupUnsubscribedPayload | WarmupStoppedPayload>) =>
+                ('key' in a.payload && a.payload.key === key) ||
+                ('keys' in a.payload && a.payload.keys.includes(key)),
+            ),
```
Comment (boxhock) [inline packages/core/bootstrap/src/lib/cache-warmer/epics.ts:193]: It seemed like this was not really working previously, so I included a fix here. Should I keep or remove?

Comment (austinborn) [reply packages/core/bootstrap/src/lib/cache-warmer/epics.ts:193]: IMO this looks good. There may be more condensed alternatives but I like the readability of it.

File: packages/sources/synthetix-debt-pool/test/integration/adapter.test.ts [hunk 2]
```diff
@@ -24,17 +26,24 @@ jest.mock('ethers', () => ({
...
+beforeAll(() => {
+  oldEnv = JSON.parse(JSON.stringify(process.env))
+  process.env.ETHEREUM_RPC_URL = 'FAKE_ETHEREUM_RPC_URL'
+  process.env.CACHE_ENABLED = 'false'
+})
+
+afterAll(() => {
+  process.env = oldEnv
+})
+
```
Comment (austinborn) [inline packages/sources/synthetix-debt-pool/test/integration/adapter.test.ts:38]: Was this change intentional for this PR?

Comment (boxhock) [reply packages/sources/synthetix-debt-pool/test/integration/adapter.test.ts:38]: Yes, this test will error out with this PR if this isn't changed :sweat_smile:

Comment (austinborn) [reply packages/sources/synthetix-debt-pool/test/integration/adapter.test.ts:38]: Yeah I see why it might cause an error without this fix...

File: packages/core/bootstrap/src/lib/cache-warmer/epics.ts [hunk 3]
```diff
@@ -177,17 +181,20 @@ export const warmupSubscriber: Epic<AnyAction, AnyAction, any, EpicDependencies>
...
             withLatestFrom(state$),
-            tap(([{ payload }, state]) => {
+            tap(([, state]) => {
               const labels = {
-                isBatched: String(
-                  !!state.cacheWarmer.subscriptions[payload.key]?.childLastSeenById,
-                ),
+                isBatched: String(!!state.cacheWarmer.subscriptions[key]?.childLastSeenById),
```
Comment (justinkaseman) [inline packages/core/bootstrap/src/lib/cache-warmer/epics.ts:197]: Oh hey, is this why the metric wasn't decrementing? Nice catch.
